### PR TITLE
CLN: Modularize code, fix count to download correct number of filings

### LINKED ISSUE #32: Instead of using a huge single class; modularise the code.
Currently, one Single class handles all the responsibilities which are not good for a codebase as it makes changing things difficult and can cause many bugs and failure.

I will design a basic architecture which is abstract away some of the responsibility.

### PR DESCRIPTION
**Pull request by jackmoody11**
- [x] closes #32, #37, #42, #44
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] added entry to CHANGELOG

### CODE REVIEW

**Comment by jackmoody11** [comment]
@coyo8 Looking back on this, it might make sense to make a single `Filings` class instead of a class for each filing type.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by jackmoody11** [comment]
@coyo8 I updated the code with the changes I mentioned earlier in this pull request.

**Comment by jackmoody11** [comment]
@coyo8 This should be merged before #52. I think I will need to make some changes to #52 before we can merge that cleanly.

File: SECEdgar/base.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            time.sleep(self.pause)
+        return self._handle_error(response)
+
+    def _validate_response(self, response):
+        """Ensures response from EDGAR is valid.
+
+        Args:
+            response (requests.response): A requests.response object.
+
+        Returns:
```
Comment (coyo8) [inline SECEdgar/base.py]: I liked this abstraction.

**Comment by coyo8** [comment]
@jackmoody11 nice work!

**Pull request merged**